lynxkite-app/web/src/workspace/NodeSearch.tsx

```diff
@@ -131,6 +131,35 @@ export default function NodeSearch(props: {
   const [selectedIndex, setSelectedIndex] = useState(0);
   const itemRefs = useRef<(HTMLButtonElement | null)[]>([]);
   const searchInputRef = useRef<HTMLInputElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  // Calculate adjusted position to keep the component visible
+  const adjustedPos = useMemo(() => {
```

Comment:
I'm not sure this needs to be a useMemo. If you just want to encapsulate the logic in a function (a good idea), you could put it in a top-level function outside of the component. I think that would be nice and readable.